action.yml

```
@@ -0,0 +1,51 @@
+# Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+# Copyright (c) 2025 Munich Quantum Software Company GmbH
+# All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+#
+# Licensed under the MIT License
+
+name: Templating
+inputs:
+  synchronize-pull-request-template:
+    description: Whether to synchronize the pull-request template
+    default: true
+  synchronize-security-policy:
+    description: Whether to synchronize the security policy
+    default: true
+  package-url:
+    description: GitHub URL of the MQT package
+    required: true
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+    - name: Install latest version of uv
+      uses: astral-sh/setup-uv@f0ec1fc3b38f5e7cd731bb6ce540c5af426746bb # v6
+    - name: Run the templating script
+      shell: bash
+      env:
+        SYNCHRONIZE_PULL_REQUEST_TEMPLATE: ${{ inputs.synchronize-pull-request-template }}
+        SYNCHRONIZE_SECURITY_POLICY: ${{ inputs.synchronize-security-policy }}
+        PACKAGE_URL: ${{ inputs.package-url }}
+      run: |
+        uv run --no-project --with jinja2 $GITHUB_ACTION_PATH/src/run.py \
+          --synchronize_pull_request_template $SYNCHRONIZE_PULL_REQUEST_TEMPLATE \
+          --synchronize_security_policy $SYNCHRONIZE_SECURITY_POLICY \
+          --package_url $PACKAGE_URL
```

```suggestion
        uv run --script $GITHUB_ACTION_PATH/src/run.py \
          --synchronize_pull_request_template $SYNCHRONIZE_PULL_REQUEST_TEMPLATE \
          --synchronize_security_policy $SYNCHRONIZE_SECURITY_POLICY \
          --package_url $PACKAGE_URL
```
Any objections against doing this and simply turning the `run.py` script into a PEP735 script with metadata? (`uv add --script run.py jinja2`)

---

Ahh... now I understand your previous comment. Done.

If I'm not mistaken, the `--script` in your suggestion is not needed, which is why I've committed this change manually.

Also, I didn't see a way to make the license tool happy with inline script metadata, which is why I'm ignoring the file now in `.license-tools-config.json`. 🤔